src/components/App.jsx

```diff
@@ -36,24 +36,27 @@ class ContactsApp extends React.Component {
           {flag('switcher') && <FlagSwitcher />}
           <ContactsSelectionBar trashAction={this.props.deleteContact} />
           <Query query={query}>
-            {({ data: contacts, fetchStatus }) => {
-              if (fetchStatus === 'loading') {
+            {({ data: contacts, fetchStatus, hasMore, fetchMore }) => {
+              if (fetchStatus === 'loaded' && hasMore === false) {
                 return (
                   <>
+                    {contacts.length >= 1 && <Header right={<Toolbar />} />}
                     <Content>
-                      <SpinnerContact
-                        size="xxlarge"
-                        loadingType="fetching_contacts"
-                      />
+                      <ContactsList contacts={contacts} />
                     </Content>
                   </>
                 )
               } else {
+                if (hasMore === true) {
```

Comment:
`if (hasMore)` ?
it feels bad/strange to fetch data in render. If you create a separate component, you could move the fetch in `componentDidUpdate`